aws/endpoints/endpoints.go

```diff
@@ -124,6 +124,46 @@ type EnumPartitions interface {
 	Partitions() []Partition
 }
 
+// RegionsForService returns a map of regions for the partition and service.
+// If either the partition or service does not exist an empty list will be
+// returned.
+//
+// This example shows how  to get the regions for DynamoDB in the AWS partition.
+//    rs := RegionsForService(endpoints.DefaultPartitions(), endpoints.AwsPartitionID, endpoints.DynamoDBServiceID)
+//
+// This is equivalent to using the partition directly.
+//    rs := endpoints.AwsPartition().Services()[endpoints.DynamoDBServiceID].Regions()
+func RegionsForService(ps []Partition, partitionID, serviceID string) map[string]Region {
```

Comment:
Should return a bool here as well for consistency and distinguishing a difference between an empty service-region map versus empty map